src/dotnet/Chat.Service/Authors.cs

```diff
@@ -36,8 +41,33 @@ public class Authors(IServiceProvider services) : DbServiceBase<ChatDbContext>(s
         if (chat == null)
             return null;
 
-        var author = await Backend.Get(chatId, authorId, AuthorsBackend_GetAuthorOption.Full, cancellationToken).ConfigureAwait(false);
-        return author.ToAuthor();
+        var authorFull = await Backend.Get(chatId, authorId, AuthorsBackend_GetAuthorOption.Full, cancellationToken).ConfigureAwait(false);
+        if (authorFull is null)
+            return null;
+
+        var author = authorFull.ToAuthor();
+        var peerUserId = authorFull.UserId;
+        var account = await Accounts.GetOwn(session, cancellationToken).ConfigureAwait(false);
+        if (account.Id != peerUserId) {
```

Comment:
Please reduce nesting and invert If condition

Reply:
resolved